### ISSUE TITLE
Circular types?

### ISSUE THREAD

**Issue by babsonmatt** [SaltyAom/mobius#1]
How would you handle something like the following, as far ordering the types? Thanks!

```gql
type Comment {
  user: User
  comment: String!
  id: ID!
}

type User {
  comments: [Comment!]!
  id: ID!
}
  ```

**Comment by SaltyAom** [comment]
This is the current limitation I have which I'm not sure how to solve it yet as TypeScript is eagerly typed.

But if really need, you need to manually type it yourself for now.

```ts
type Comment = {
    comment: string
    id: string
    user: User
}

type User = {
    comments: [Comment]
    id: string
}

const client = new Mobius<typeof typeDefs, {
    User: User
    Comment: Comment
}>()
```

Please keep this opened until I find a way to solve this

**Comment by ggrandi** [comment]
I saw this project on twitter and thought it looked really cool so I tried investigating a proof of concept on how to solve this circular types issue.

I ended up coming up with this: https://tsplay.dev/mLX7vm

It uses the fact that interfaces can reference the `this` type to have recursion. I haven't yet tried to integrate it with the rest of mobius but the concept seems to work well.

I'd be happy to try to create a PR to make it work
